### PR TITLE
[#643] AMQP adapter CLI

### PR DESCRIPTION
**Pull request by Alfusainey**
@sophokles73 This integrates the command-line for the amqp adapter into the hono `cli` module.

Signed-off-by: Alfusainey Jallow <alf.jallow@gmail.com>

### CODE REVIEW

File: cli/pom.xml [hunk 1]
```diff
@@ -29,6 +29,11 @@
     <description>Hono Command line Interface</description>
 
     <dependencies>
+        <dependency>
+            <groupId>commons-cli</groupId>
+            <artifactId>commons-cli</artifactId>
```
Comment (sophokles73) [inline cli/pom.xml]: I do not really understand what we need this for. Spring Boot supports binding command line params to fields or beans out-of-the-box ...

**Comment by Alfusainey** [comment]
thanks... 183c8c2 removes the `commons-cli` dependency

File: cli/src/main/resources/application.yml [hunk 2]
```diff
@@ -44,3 +44,9 @@ hono:
   client:
     hostnameVerificationRequired: false
     trustStorePath: target/config/hono-demo-certs-jar/trusted-certs.pem
+
+---
+
+spring:
+  profiles: amqp-adapter-cli
+
```
Comment (sophokles73) [inline cli/src/main/resources/application.yml]: missing EOL

**Comment by Alfusainey** [comment]
@sophokles73: PR updated

File: cli/src/main/resources/application.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ spring:
   jmx:
     enabled: false
   profiles:
-    active: receiver,ssl
+    active: receiver,ssl,amqp-adapter-cli
```
Comment (sophokles73) [inline cli/src/main/resources/application.yml]: My understanding is that this property defines the profiles to be active by default. FMPOV it doesn't make sense to activate receiver together with adapter client, does it?

File: cli/src/main/resources/application.yml [hunk 4]
```diff
@@ -44,3 +44,9 @@ hono:
   client:
     hostnameVerificationRequired: false
     trustStorePath: target/config/hono-demo-certs-jar/trusted-certs.pem
+
+---
+
+spring:
+  profiles: amqp-adapter-cli
```
Comment (sophokles73) [inline cli/src/main/resources/application.yml]: you should add some defaults to be used when running the application as AMQP adapter client, e.g. the default port to connect to ...

**Comment by sophokles73** [comment]
@Alfusainey can you also paste a command line here for how you would actually run the client?

File: cli/src/main/resources/application.yml (hunk 3 above)
Comment (Alfusainey) [reply cli/src/main/resources/application.yml]: actually, i was wondering about this... i want to completely remove the `amqp-adapter-cli` profile and activate it using the `--spring.profiles.active` switch. WDYT?

File: cli/src/main/resources/application.yml (hunk 4 above)
Comment (Alfusainey) [reply cli/src/main/resources/application.yml]: ok

**Comment by Alfusainey** [comment]
> @Alfusainey can you also paste a command line here for how you would actually run the client?

spring-boot:run -Drun.arguments=--message.address=telemetry/DEFAULT_TENANT/4712,--amqp.host=localhost,--amqp.port=4040,--username=sensor1@DEFAULT_TENANT,--password=hono-secret,--message='{"temp": 5}'

yes, i agree.. its a bit verbose :-)

**Comment by sophokles73** [comment]
@Alfusainey you could put all of those values as default values into the yaml file ...

File: cli/src/main/resources/application.yml (hunk 3 above)
Comment (sophokles73) [reply cli/src/main/resources/application.yml]: >  i want to completely remove the amqp-adapter-cli profile and activate it using the --spring.profiles.active switch. WDYT?

FMPOV you simply need to remove `amqp-adapter-cli` from this line and then you will be able to do exactly what you want to do, i.e. using the `--spring.profiles.active` switch ...

**Comment by Alfusainey** [comment]
> @Alfusainey you could put all of those values as default values into the yaml file ...

sure.

**Comment by sophokles73** [comment]
This actually starts up a maven build, which is quite slow. A better alternative is to run the jar file created in the target folder:
```
java target/hono-cli-0.7-SNAPSHOT.jar --message.address=telemetry/DEFAULT_TENANT/4712 --amqp.host=localhost --amqp.port=4040 --username=sensor1@DEFAULT_TENANT --password=hono-secret --message='{"temp": 5}'
```

**Comment by Alfusainey** [comment]
@sophokles73: with 5ad5e8d defaults values for the client is added in the YAML file

**Comment by Alfusainey** [comment]
> This actually starts up a maven build, which is quite slow

okay, i will use the bundled JAR archive in the example docs. the `amqp-adapter-cli` profile is not active by default. how about having users add it via the `--spring.profiles.active` switch?

**Pull request merged**